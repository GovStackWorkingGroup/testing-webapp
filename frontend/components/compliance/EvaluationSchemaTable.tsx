import useTranslations from '../../hooks/useTranslation';
import { DataType } from '../../service/types';
import Table from '../table/Table';

const EvaluationSchemaTable = () => {
  const { format } = useTranslations();

  const headers = [
    'evaluation_schema.compliance.label',
    'evaluation_schema.requirement.label',
    'evaluation_schema.level_1.label',
    'evaluation_schema.level_2.label',
  ];

  const data: DataType = {
    rows: [
      {
        cell: [
          {
            value: format('table.deployment_compliance.label'),
          },
          {
            value: format(
              'evaluation_schema.deployability_via_container.label'
            ),
          },
          { value: format('table.deployable.label') },
          { value: format('table.N/A.label') },
        ],
      },
      {
        cell: [
          {
            value: format('table.requirement_specification.label'),
          },
          {
            values: [
              {
                value: format(
                  'evaluation_schema.fulfillment_of_required_cc.label'
                ),
              },
              {
                value: format(
                  'evaluation_schema.fulfillment_of_required_functional_requirements.label'
                ),
              },
            ],
          },
          {
            values: [
              {
                value: format('evaluation_schema.over50.label'),
              },
              {
                value: format('evaluation_schema.over90.label'),
              },
            ],
          },
          {
            values: [
              {
                value: format('evaluation_schema.over50.label'),
              },
              {
                value: format('evaluation_schema.over90.label'),
              },
            ],
          },
        ],
      },
      {
        cell: [
          {
            value: format('table.interface_compliance.label'),
          },
          {
            values: [
              {
                value: format(
                  'evaluation_schema.fulfillment_of_service_api.label'
                ),
              },
              {
                value: format(
                  'evaluation_schema.fulfillment_of_required_api.label'
                ),
              },
            ],
          },
          {
            values: [
              {
                value: format('evaluation_schema.equal_or_more_than_1.label'),
              },
              {
                value: format('evaluation_schema.optional.label'),
              },
            ],
          },
          {
            values: [
              {
                value: format('evaluation_schema.all.label'),
              },
              {
                value: format('evaluation_schema.all.label'),
              },
            ],
          },
        ],
      },
    ],
  };

  return <Table data={data} headers={headers} isEvaluationSchema={true} />;
};

export default EvaluationSchemaTable;
