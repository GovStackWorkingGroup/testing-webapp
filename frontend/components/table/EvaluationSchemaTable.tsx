import useTranslations from '../../hooks/useTranslation';
import Table from './Table';

const EvaluationSchemaTable = () => {
  const { format } = useTranslations();
  const data = {
    headers: [
      format('evaluation_schema.compliance.label'),
      format('evaluation_schema.requirement.label'),
      format('evaluation_schema.level_1.label'),
      format('evaluation_schema.level_2.label'),
    ],
    rows: [
      {
        cell: [
          {
            value: format('table.deployment_compliance.label'),
          },
          {
            value: format(
              'evaluation_schema.deployability_via_container.label'
            ),
          },
          { value: 'checked' },
          { value: 'checked' },
        ],
      },
      {
        cell: [
          {
            value: format('table.interface_compliance.label'),
          },
          {
            values: [
              {
                value: format(
                  'evaluation_schema.fulfillment_of_service_api.label'
                ),
              },
              {
                value: format(
                  'evaluation_schema.fulfillment_of_required_api.label'
                ),
              },
            ],
          },
          {
            values: [
              {
                value: format('evaluation_schema.equal_or_more_than_1.label'),
              },
              {
                value: format('evaluation_schema.optional.label'),
              },
            ],
          },
          {
            values: [
              {
                value: format('evaluation_schema.all.label'),
              },
              {
                value: format('evaluation_schema.all.label'),
              },
            ],
          },
        ],
      },
      {
        cell: [
          {
            value: format('evaluation_schema.requirement_specification.label'),
          },
          {
            values: [
              {
                value: format(
                  'evaluation_schema.fulfillment_of_required_key.label'
                ),
              },
              {
                value: format(
                  'evaluation_schema.fulfillment_of_required_functional_requirements.label'
                ),
              },
              {
                value: format(
                  'evaluation_schema.fulfillment_of_required_cross_cutting_requirements.label'
                ),
              },
            ],
          },
          {
            values: [
              {
                value: format('evaluation_schema.all.label'),
              },
              {
                value: format('evaluation_schema.equal_or_more_than_1.label'),
              },
              { value: format('evaluation_schema.equal_or_more_than_1.label') },
            ],
          },
          {
            values: [
              {
                value: format('evaluation_schema.all.label'),
              },
              {
                value: format('evaluation_schema.all.label'),
              },
              {
                value: format('evaluation_schema.all.label'),
              },
            ],
          },
        ],
      },
    ],
  };

  return <Table data={data} />;
};

export default EvaluationSchemaTable;
