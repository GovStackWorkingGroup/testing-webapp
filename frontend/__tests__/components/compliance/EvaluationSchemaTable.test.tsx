import EvaluationSchemaTable from '../../../components/compliance/EvaluationSchemaTable';
import { render } from '../../test-utils/test-utils';

describe('Unit tests for EvaluationSchemaTable component:', () => {
  it('should render the component and match snapshot', () => {
    const { container } = render(<EvaluationSchemaTable />);

    expect(container).toMatchSnapshot();
  });
});
