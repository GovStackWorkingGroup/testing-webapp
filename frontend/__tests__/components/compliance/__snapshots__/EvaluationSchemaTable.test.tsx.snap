// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for EvaluationSchemaTable component: should render the component and match snapshot 1`] = `
<div>
  <table
    class="main-table"
  >
    <thead>
      <tr>
        <th
          class=""
        >
          Compliance
        </th>
        <th
          class=""
        >
          Requirement
        </th>
        <th
          class=""
        >
          Level 1
        </th>
        <th
          class=""
        >
          Level 2
        </th>
      </tr>
    </thead>
    <tbody
      id="scrollableDiv"
    >
      <tr
        class="border"
      >
        <td
          class="border"
        >
          Deployment Compliance
        </td>
        <td
          class="border"
        >
          Deployability via container
        </td>
        <td
          class="border"
        >
          Deployable
        </td>
        <td
          class="border"
        >
          N/A
        </td>
      </tr>
      <tr
        class="border"
      >
        <td
          class="border"
        >
          Requirement Specification Compliance
        </td>
        <td
          class="td-row-details"
        >
          <table
            class="main-table"
          >
            <tbody
              class=""
            >
              <tr>
                <td
                  class="border-none border-bottom"
                >
                  Fulfillment of REQUIRED Key Digital Functionalities stated in the respective BB specifications
                </td>
              </tr>
              <tr>
                <td
                  class="border-none border-bottom"
                >
                  Fulfillment of REQUIRED cross-cutting and functional requirements stated in the respective BB specifications
                </td>
              </tr>
              <tr>
                <td
                  class="border-none"
                >
                  Fulfillment of REQUIRED cross-cutting requirements stated in the Architecture BB specifications
                </td>
              </tr>
            </tbody>
          </table>
        </td>
        <td
          class="td-row-details"
        >
          <table
            class="main-table"
          >
            <tbody
              class=""
            >
              <tr>
                <td
                  class="border-none border-bottom"
                >
                  All
                </td>
              </tr>
              <tr>
                <td
                  class="border-none border-bottom"
                >
                  ≥ 1
                </td>
              </tr>
              <tr>
                <td
                  class="border-none"
                >
                  ≥ 1
                </td>
              </tr>
            </tbody>
          </table>
        </td>
        <td
          class="td-row-details"
        >
          <table
            class="main-table"
          >
            <tbody
              class=""
            >
              <tr>
                <td
                  class="border-none border-bottom"
                >
                  All
                </td>
              </tr>
              <tr>
                <td
                  class="border-none border-bottom"
                >
                  All
                </td>
              </tr>
              <tr>
                <td
                  class="border-none"
                >
                  All
                </td>
              </tr>
            </tbody>
          </table>
        </td>
      </tr>
      <tr
        class="border"
      >
        <td
          class="border"
        >
          API Compliance
        </td>
        <td
          class="td-row-details"
        >
          <table
            class="main-table"
          >
            <tbody
              class=""
            >
              <tr>
                <td
                  class="border-none border-bottom"
                >
                  Fulfillment of Service API requirements
                </td>
              </tr>
              <tr>
                <td
                  class="border-none"
                >
                  Fulfillment of REQUIRED API related requirements in the Architecture BB specifications
                </td>
              </tr>
            </tbody>
          </table>
        </td>
        <td
          class="td-row-details"
        >
          <table
            class="main-table"
          >
            <tbody
              class=""
            >
              <tr>
                <td
                  class="border-none border-bottom"
                >
                  ≥ 1
                </td>
              </tr>
              <tr>
                <td
                  class="border-none"
                >
                  Optional
                </td>
              </tr>
            </tbody>
          </table>
        </td>
        <td
          class="td-row-details"
        >
          <table
            class="main-table"
          >
            <tbody
              class=""
            >
              <tr>
                <td
                  class="border-none border-bottom"
                >
                  All
                </td>
              </tr>
              <tr>
                <td
                  class="border-none"
                >
                  All
                </td>
              </tr>
            </tbody>
          </table>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
